examples/train_resnet.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+import os
+import numpy as np
+import random
+from PIL import Image
+
+from tinygrad.tensor import Device
+from extra.utils import get_parameters
+from extra.training import train, evaluate
+from extra.resnet import ResNet18, ResNet34, ResNet50
+from tinygrad.optim import Adam
+from test.test_mnist import fetch_mnist
+
+from tinygrad.optim import Adam
+
+class ComposeTransforms:
+  def __init__(self, trans):
+    self.trans = trans
+
+  def __call__(self, x):
+    for t in self.trans:
+      x = t(x)
+    return x
+
+if __name__ == "__main__":
+  model = ResNet18(num_classes=10, pretrained=True)
+
+  X_train, Y_train, X_test, Y_test = fetch_mnist()
+  lr = 0.003
+  transform = ComposeTransforms([
```

Comment:
Perhaps just preapply this? It lets you remove the complexity in the trainer

Clever idea about big mnist btw.